### ISSUE TITLE
[lib] Parameter names in synopses

### ISSUE THREAD

**Issue by jensmaurer** [cplusplus/draft#3255]
While we strive to keep the declaration in a synopsis the same as the one in an `\itemdecl`, we're inconsistent (at least) for copy/move constructors and copy/move assignment operators.

Personally, I feel that copy/move operations speak for themselves through the declaration and don't need a distracting parameter name in the synopsis, in particular if the function is defaulted or deleted. (We need one for the `\itemdecl`, of course, because we need to talk about the source of the copy/move.)

See #3230 for a specific case.

**Comment by jwakely** [comment]
I agree 100% with removing parameter names from defaulted and deleted functions, and from special member functions where we declare them in the synopsis but don't have an `\itemdecl` for them.

**Comment by jensmaurer** [comment]
Editorial meeting: \itemdecl should not introduce primary information; it's just a heading. Parameter names might meet that bar. Arbitrarily picking a name seems reasonable. Investigate prevailing existing style. Suggest to use "other" consistently.
We allow a difference for itemdecl vs. synopsis, but if a parameter name is present, it should be the same in both places. Having a parameter name that is an abbreviation of the type name is not useful in the synopsis.

**Comment by jwakely** [comment]
[counted.iterator] has this in the class synopsis:

    constexpr counted_iterator(I x, iter_difference_t<I> n);

[counted.iter.const] has a different parameter name:

    constexpr counted_iterator(I i, iter_difference_t<I> n);
